src/prisma/schedule.ts

```diff
@@ -37,3 +39,108 @@ async function initiateSchedule(scheduleyear: number) {
   });
   return schedule;
 }
+
+async function updateCurrentSchedule(
+  scheduleId: any,
+  courses: CourseSectionInput[]
+) {
+  // Simply creates a new schedule instead of updating the current schedule
+
+  try {
+    const currentSchedule = await prisma.schedule.findFirst({
+      where: {
+        id: parseInt(scheduleId),
+      },
+    });
+
+    if (currentSchedule) {
+      // Delete course sections linked to original schedule
+      await prisma.courseSection.deleteMany({
+        where: {
+          scheduleId: parseInt(scheduleId),
+        },
+      });
+
+      // Delete original schedule
+      await prisma.schedule.delete({
+        where: {
+          id: parseInt(scheduleId),
+        },
+      });
```

Comment:
I don't think we need to delete the schedule entry as the behaviour is:
>   ID of the schedule to update. If not given, the current schedule will be updated.
We're deleting the schedule in the sense of the contents not the entry itself.
I don't think this will be breaking anything but I think its good to double check here.